command/node_status.go

```diff
@@ -515,6 +536,52 @@ func (c *NodeStatusCommand) outputNodeVolumeInfo(node *api.Node) {
 	c.Ui.Output(formatList(output))
 }
 
+func (c *NodeStatusCommand) outputNodeCSIVolumeInfo(client *api.Client, node *api.Node, runningAllocs []*api.Allocation) {
+	c.Ui.Output(c.Colorize().Color("\n[bold]CSI Volumes"))
+
+	// Duplicate nodeCSIVolumeNames to sort by name but also index volume names to ids
+	var names []string
+	volNames := map[string]string{}
+	for _, alloc := range runningAllocs {
+		tg := alloc.GetTaskGroup()
+		if tg == nil || len(tg.Volumes) == 0 {
+			continue
+		}
+
+		for _, v := range tg.Volumes {
+			names = append(names, v.Name)
+			volNames[v.Source] = v.Name
+		}
+	}
+	sort.Strings(names)
+
+	// Fetch the volume objects with current status
+	// Ignore an error, all we're going to do is omit the volumes
+	volumes := map[string]*api.CSIVolumeListStub{}
+	vs, _ := client.Nodes().CSIVolumes(node.ID, nil)
+	for _, v := range vs {
+		n := volNames[v.ID]
+		volumes[n] = v
+	}
+
+	// Output the volumes in name order
+	output := make([]string, 0, len(names)+1)
+	output = append(output, "ID|Name|Plugin ID|Schedulable|Access Mode")
+	for _, name := range names {
+		v := volumes[name]
+		output = append(output, fmt.Sprintf(
+			"%s|%s|%s|%t|%s",
+			v.ID,
+			name,
+			v.PluginID,
+			v.Schedulable,
```

Comment:
> For consideration: we show `Schedulable` in the verbose status, which is maybe a bit confusing in this context?

There might be value in having a consistent representation even if it's not super _useful_ in this context. Do we want to have the `Provider` here though?

Reply:
whoops, fixed in 32e0ad21f